haystack/modeling/training/base.py

```diff
@@ -829,7 +829,23 @@ def __init__(
         from_step=from_step, global_step=global_step,
         evaluator_test=evaluator_test, disable_tqdm=disable_tqdm,
         max_grad_norm=max_grad_norm)
-        self.teacher_model = teacher_model
+        self.loss = DataParallel(DistillationLoss(model, teacher_model, device))
+        if torch.cuda.device_count() > 1:
```

Comment:
I think in general we would like to support cases where a user has access to GPUs but still might not want to use them. Please check whether `len(self.devices) > 1:`makes more sense here or as in other modules, use `self.devices, _ = initialize_device_settings(use_cuda=use_gpu, multi_gpu=False)`.

Reply:
I would agree that this functionality would be useful.
However, the current behaviour is actually more in line with the other implementations. Setting the parameter `multi_gpu` does not have the expected outcome in `self.devices, _ = initialize_device_settings(use_cuda=use_gpu, multi_gpu=False)`. This is due to [this line](https://github.com/deepset-ai/haystack/blob/7d769d8bf14a72801c695728c87f5f4eee0e9b87/haystack/modeling/model/optimization.py#L305) which ignores the `devices` parameter and also uses the condition `len(self.devices) > 1`.

For this reason, I would think that it makes more sense to add the functionality to disable using multiple GPUs separately when also implementing this feature for `train` and `distil_prediction_layers_from` and changing the functionality of [`optimize_model`](https://github.com/deepset-ai/haystack/blob/7d769d8bf14a72801c695728c87f5f4eee0e9b87/haystack/modeling/model/optimization.py#L270).

Reply:
Alright agreed. Let's not address that behavior in this PR.

Reply:
@MichelBartels the type annotation needs to be `torch.device` though :)

Reply:
Yes, that's right unfortunately. Working on it here: https://github.com/deepset-ai/haystack/pull/1976#issuecomment-1020913754